adapter/ipListChecker/adapter.go

```diff
@@ -16,73 +16,96 @@ package ipListChecker
 
 import (
 	"errors"
+	"fmt"
+	"net"
+	"net/http"
 	"net/url"
 	"time"
 
-	"istio.io/mixer/pkg/adapter"
+	"github.com/golang/protobuf/proto"
+
+	"istio.io/mixer/pkg/aspect/listChecker"
+	"istio.io/mixer/pkg/aspectsupport"
+)
+
+const (
+	// ImplName is the canonical name of this implementation
+	ImplName = "istio/IPListChecker"
 )
 
-// AdapterConfig is used to configure a adapter.
-type AdapterConfig struct {
+// Register registration entry point
+func Register(r aspectsupport.Registry) {
+	r.RegisterCheckList(&adapterState{})
 }
 
 type adapterState struct{}
 
-// NewAdapter returns a Adapter
-func NewAdapter() adapter.Adapter {
-	return &adapterState{}
-}
-
 func (a *adapterState) Name() string {
-	return "IPListChecker"
+	return ImplName
 }
 
 func (a *adapterState) Description() string {
 	return "Checks whether an IP address is present in an IP address list."
```

Comment:
Yes, I think it makes sense.
I am a bit conflicted on imposing so much cognitive overhead on adapter authors though.